### ISSUE TITLE
EvaluatableExpression API does not allow extensinos to return structured objects

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#91408]
Ref: #91268

Currently the evaluatableExpression API only allows extensions to return string results to VS Code.
It would be nice if it was somehow possible to return structured objects. Or potentially lazily fetched objects that VS Code could fetch from the Debug Adapter.

**Comment by weinand** [comment]
The contract between the hover and the debugger is the "evaluate" request which takes an expression string. I do not see how we could use a structured object here. In addition the EvaluatableExpression API is orthogonal to debuggers, so the EvaluatableExpression cannot make any assumptions about the debugger that evaluates the expression.

@isidorn what exactly is your use case?

**Comment by isidorn** [comment]
My usage scenario is 
`var t = { name: "isidor", lastName: "nikolic"} `
User hovers over t, the extension can provide an expandable object as the result of the EvaluatableExpression such that it is shown in the hover.

I see the evaluatableExpression as an alternative way of saying "let the extension do the evaluation for the hover". And this is currently the missing piece imho.

If the extension cannot make assumptions about the debugger it shuold still be able to send an object like structure (which does not work for deeply nested objects so this would have to be lazy somehow).

**Comment by weinand** [comment]
The purpose of the API is to return an expression, not to do the evaluation. Your example is already doable by returning the structured object as a string. The subsequent evaluate will do exactly what you asking for.

**Comment by connor4312** [comment]
For js I like the current implementation as it's opaque to the debug adapter and requires no changes to the adapter as languages implement the interface and improve. Of course they could pack information into the "expression". I could see more cases where a compiled language may need to provide additional aside from an 'expression' to correlate an identifier in the source code to runtime data--though they could of course pack information into the string. Do we have requests from other debug adapter implementors to provide more than a string?

**Comment by weinand** [comment]
@connor4312 no, we do not have requests for more than a string and all debug adapter implementors are happy with the new API because they are able solve their issues.

**Comment by isidorn** [comment]
I have verified that returning `{ name: "isidor", lastName: "nikolic"}` as the evaluatbleExpression nicely evaluates it if the debugger supports evaluation on hover.
Due to that I think this issue is invalid and it can be closed. @weinand feel free to do that

However the potential confusion still remains, but I have commented on that in a separate issues - that we should clarify that this api is really about expressions, not the results of evaluations.
